org.adempiere.base/src/org/compiere/model/GridTable.java

```diff
@@ -3216,14 +3230,15 @@ private void doRun() {
 			{
 				if (! DBException.isTimeout(e))
 					throw new AdempiereException(e);
+				if (log.isLoggable(Level.INFO)) log.info("Query timed out -> " + m_SQL);
 			}
 			finally
 			{
 				close();
 			}
 			
 			// Background loading without initial rowCount, inform final loaded rows
-			if (m_rowCountTimeout && m_sort.size() > 0)
+			if (backgroundLoad && m_sort.size() > 0)

```

Comment:
this is changing the original logic, perhaps it should be `((m_rowCountTimeout || backgroundLoad) && m_sort.size() > 0)` instead.

Reply:
> this is changing the original logic, perhaps it should be `((m_rowCountTimeout || backgroundLoad) && m_sort.size() > 0)` instead.

I noticed that informing the final rows is just needed when loading in background, not when the count times out.

The scenario is like this:
* I opened "Change Audit" window with 27 million records - the count times out showing "Records = ?" in the Find dialog
* then I search for changes on a table which returns few records, the query is very fast and doesn't require background loading
* in this case, there is no need to inform at the end the final loaded rows, because there was no background load, the system shows correctly the number of records found